CMakeLists.txt

```diff
@@ -199,13 +199,13 @@ endif()
 ####
 # Setup the compiler options
 
-# set c++ standard to c++11.
+# set c++ standard to c++14.
 # Note: not working on CMake 2.8. We assume that user has
-#       a compiler with C++11 support.
+#       a compiler with C++14 support.
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-message(STATUS "C++11 support has been enabled by default.")
+message(STATUS "C++14 support has been enabled by default.")
```

Comment:
@Randl It'd be helpful if you mention the minimum versions of supported compilers in `/README.md`.

Talking about nyanp's comment
> but it's ok for me if we don't use C++14 outside of xtensor, because xtensor supports at least msvc2015 update2.

I wonder if people would follow the mentioned wish/rule? and it reminds me of my long-ago memory that my high school classmates smoke puff on a balcony.